src/common/server.ts

```diff
@@ -83,6 +85,28 @@ async function getRuffVersion(executable: string): Promise<VersionInfo> {
   return { major, minor, patch };
 }
 
+/**
+ * Validate and log executable from given path.
+ */
+async function validateUsingExecutable(executable: string, strategy: string) {
+  try {
+    const ruffVersion = await getRuffVersion(executable);
+    if (!supportsExecutable(ruffVersion)) {
+      var message = `Skip unsupported executable from ${strategy}: ${executable}`;
+      message += `(Reqiuired at least ${versionToString(
+        MINIMUM_SUPPORTED_EXECUTABLE_VERSION,
+      )}, but found ${versionToString(ruffVersion)} instead)`;
+      traceError(message);
+      return false;
+    }
+    traceInfo(`Using ${strategy}: ${executable}`);
+    return true;
+  } catch (ex) {
+    traceInfo(`Skip invalid executable from ${strategy}: ${executable}`);
```

Comment:
Should we include the exception message as well? Are we able to capture the one in #426 ?

Reply:
Yes, makes sense. Since `ex` maybe multi line error, what is correct format to put it in there? I don't think put it in parantheses be a good style.
```js
traceInfo(`Skip invalid executable from ${strategy}: ${executable}:\n${ex}`);
```
is this correct to you?

Reply:
I think this should be fine but I'd need to play around with it to check how it renders in the output window. I can do that when I'm testing before merging the PR.